lib/spack/llnl/util/filesystem.py

```diff
@@ -325,6 +326,12 @@ def ancestor(dir, n=1):
     return parent
 
 
+def assuredir(path):
```

Comment:
Can this be removed? As far as I can tell, this function is no longer called based on the updates in this PR.